include/h2o.h

```diff
@@ -628,6 +638,10 @@ typedef struct st_h2o_header_t {
      * value of the header
      */
     h2o_iovec_t value;
+    /**
+     * If non-NULL, contains a bitfield indicating whether a character in @name was upper case
+     */
+    h2o_str_case_t *orig_hname_case;
```

Comment:
I am unsure if we want to save memory at the cost of spending extra cycles for building the header in the original case. Unless you have a need, lets just store a `char *` that points to the characters in the original case.